Gradle copying artifacts for native build failed with Permission denied

### Describe the bug

Similar to https://github.com/quarkusio/quarkus/issues/33842, gradle QarkusBuildTask fails to copy readonly file from the native build source dir (`nativeImageSourceJarDirName() + "/**"))`).

https://github.com/quarkusio/quarkus/blob/382cf447d81f831e8830b5ffe2e2e512803d91ac/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusBuild.java#L340-L361



### Expected behavior

Build is sucessful

### Actual behavior

Build failed

```
> Task :quarkusBuild FAILED
Caching disabled for task ':quarkusBuild' because:
  Build cache is disabled
Task ':quarkusBuild' is not up-to-date because:
  Task has failed previously.
Removing output files and directories (provide a clean state).
Synchronizing Quarkus native build from /home/user/source/quarkus/test/quarkus-amazon-services-crt-client-issue/build/quarkus-build/app/quarkus-app and /home/user/source/quarkus/test/quarkus-amazon-services-crt-client-issue/build/quarkus-build/dep into /home/user/source/quarkus/test/quarkus-amazon-services-crt-client-issue/build/quarkus-app
file or directory '/home/user/source/quarkus/test/quarkus-amazon-services-crt-client-issue/build/quarkus-build/app/quarkus-app', not found
Copying remaining Quarkus application artifacts for native build from /home/user/source/quarkus/test/quarkus-amazon-services-crt-client-issue/build/quarkus-build/app into /home/user/source/quarkus/test/quarkus-amazon-services-crt-client-issue/build
Build c9733143-e847-4e44-9888-17992dff296e is closed

[Incubating] Problems report is available at: file:///home/user/source/quarkus/test/quarkus-amazon-services-crt-client-issue/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':quarkusBuild'.
> Could not copy file '/home/user/source/quarkus/test/quarkus-amazon-services-crt-client-issue/build/quarkus-build/app/quarkus-amazon-services-crt-client-issue-1.0.0-SNAPSHOT-native-image-source-jar/libaws-crt-jni.so' to '/home/user/source/quarkus/test/quarkus-amazon-services-crt-client-issue/build/quarkus-amazon-services-crt-client-issue-1.0.0-SNAPSHOT-native-image-source-jar/libaws-crt-jni.so'.
   > /home/user/source/quarkus/test/quarkus-amazon-services-crt-client-issue/build/quarkus-amazon-services-crt-client-issue-1.0.0-SNAPSHOT-native-image-source-jar/libaws-crt-jni.so (Permission denied)

```

### How to Reproduce?

More info and a reproducer at https://github.com/quarkiverse/quarkus-amazon-services/issues/1694

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Mandrel or GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

The file is created and set readonly by a GraalVM feature.

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusBuild.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusBuild.java
@@ -351,13 +351,16 @@ public abstract class QuarkusBuild extends QuarkusBuildTask {
                     jarType(), sourceDir, buildDir);
         }
         getFileSystemOperations().copy(
-                copy -> copy.into(buildDir).from(sourceDir)
-                        .include(QUARKUS_ARTIFACT_PROPERTIES,
-                                nativeRunnerFileName(),
-                                runnerJarFileName(),
-                                "jib-image*",
-                                NATIVE_SOURCES,
-                                nativeImageSourceJarDirName() + "/**"));
+                copy -> {
+                    copy.eachFile(new CopyActionDeleteNonWriteableTarget(buildDir.toPath()));
+                    copy.into(buildDir).from(sourceDir)
+                            .include(QUARKUS_ARTIFACT_PROPERTIES,
+                                    nativeRunnerFileName(),
+                                    runnerJarFileName(),
+                                    "jib-image*",
+                                    NATIVE_SOURCES,
+                                    nativeImageSourceJarDirName() + "/**");
+                });
     }
 
     private String expandConfigurationKey(String shortKey) {

